pkg/oc/cli/admin/groups/users/add.go

```diff
@@ -0,0 +1,146 @@
+package users
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+	"k8s.io/kubernetes/pkg/printers"
+
+	userv1 "github.com/openshift/api/user/v1"
+	userv1typedclient "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+)
+
+const AddRecommendedName = "add-users"
+
+var (
+	addLong = templates.LongDesc(`
+		Add users to a group.
+
+		This command will append unique users to the list of members for a group.`)
+
+	addExample = templates.Examples(`
+		# Add user1 and user2 to my-group
+	%[1]s my-group user1 user2`)
+)
+
+type AddUsersOptions struct {
+	GroupModificationOptions *GroupModificationOptions
+}
+
+func NewCmdAddUsers(name, fullName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &AddUsersOptions{
+		GroupModificationOptions: NewGroupModificationOptions(streams),
+	}
+	cmd := &cobra.Command{
+		Use:     name + " GROUP USER [USER ...]",
+		Short:   "Add users to a group",
+		Long:    addLong,
+		Example: fmt.Sprintf(addExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	o.GroupModificationOptions.PrintFlags.AddFlags(cmd)
+	kcmdutil.AddDryRunFlag(cmd)
+
+	return cmd
+}
+
+func (o *AddUsersOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	return o.GroupModificationOptions.Complete(f, cmd, args)
+}
+
+func (o *AddUsersOptions) Run() error {
+	group, err := o.GroupModificationOptions.GroupClient.Groups().Get(o.GroupModificationOptions.Group, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	existingUsers := sets.NewString(group.Users...)
+	for _, user := range o.GroupModificationOptions.Users {
+		if existingUsers.Has(user) {
+			continue
+		}
+
+		group.Users = append(group.Users, user)
+	}
+
+	if !o.GroupModificationOptions.DryRun {
+		group, err = o.GroupModificationOptions.GroupClient.Groups().Update(group)
+		if err != nil {
+			return err
+		}
+	}
+
+	return o.GroupModificationOptions.PrintObj("added", group)
```

Comment:
Hm, this does not look right. Is this using our printers?

Reply:
It uses a helper, which calls `ToPrinter` to inject appropriate message, either added or removed and then calls `PrintObj`.